lib/openshift/machine.rb

```diff
@@ -28,7 +28,9 @@ def phase(user: nil, cached: true, quiet: false)
 
     def instance_state(user: nil, cached: true, quiet: false)
       raw_resource(user: user, cached: cached, quiet: quiet).
-          dig('status', 'providerStatus', 'instanceState')
+          dig('status', 'providerStatus', 'instanceState') ||
+      raw_resource(user: user, cached: cached, quiet: quiet).
+          dig('status', 'providerStatus', 'vmState')
```

Comment:
LGTM. Perhaps we can shorten it as
```
rr = raw_resource(user: user, cached: cached, quiet: quiet)
rr.dig('status', 'providerStatus', 'instanceState') ||
rr.dig('status', 'providerStatus', 'vmState')
```

Reply:
Thanks @liangxia  , add as suggessted , validated the change looks good .
     [10:03:22] INFO> Exit Status: 0
    Then the expression should be true> cb.machines.select{|m|m.instance_state == m.annotation_instance_state}.count == cb.machines.count     # features/step_definitions/common.rb:133
waiting for operation up to 3600 seconds..
waiting for operation up to 3600 seconds..
      [10:03:22] INFO> === After Scenario: Verify all machine instance-state should be consistent with their providerStats.instanceState ===
      [10:03:22] INFO> Shell Commands: rm -r -f -- /root/workdir/c92ee1977c9c-
      
      [10:03:22] INFO> Exit Status: 0
      [10:03:22] INFO> === End After Scenario: Verify all machine instance-state should be consistent with their providerStats.instanceState ===

1 scenario (1 passed)
3 steps (3 passed)
0m13.642s
[10:03:22] INFO> === At Exit ===